indra/newview/llvoavatarself.cpp

```diff
@@ -1292,12 +1292,24 @@ U32 LLVOAvatarSelf::getNumWearables(LLAvatarAppearanceDefines::ETextureIndex i)
 	return gAgentWearables.getWearableCount(type);
 }
 
-void LLVOAvatarSelf::postMotionUpdate() 
+void LLVOAvatarSelf::updateMotions(LLCharacter::e_update_t update_type)
 {
+    LLCharacter::updateMotions(update_type);
+
+    // post motion update
+    if (!mAttachmentUpdateEnabled)
+    {
+        return;
+    }
+    U64 now = LLFrameTimer::getTotalTime();
+    if (now > mAttachmentUpdateExpiry)
+    {
+        return;
+    }
+    mAttachmentUpdateExpiry = now + mAttachmentUpdatePeriod;
+
```

Comment:
Easy and fast early exits are above, more complicated early exit is below.

We're also switching to use an expiry pattern for deciding when to send update rather than use an `LLFrameTimer` instance.